src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int INCREASE_RATE = 2;
+
+    private Node<K, V>[] table;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = createNode(key, value);
+        addToTable(node, table, true);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = calculateIndexByHash(key == null ? 0 : key.hashCode());
```

Comment:
create method for that
```suggestion
        int index = getIndex(key);
```